stl/src/syserror.cpp

```diff
@@ -4,11 +4,14 @@
 // system_error message mapping
 
 #include <cstdlib>
-#include <memory>
 #include <system_error>
 
 #include <Windows.h>
 
+#ifdef _DLL
+#include "syserror_import_lib.cpp"
+#endif

```

Comment:
There are calls to these from within "P" itself; we need the import lib functions provided when compiling "P" so that it links.

Reply:
Can we modify the build system accordingly to build this into both the DLL and the import lib? Having a `.cpp` file include another is weird and icky.

Now I see that https://github.com/microsoft/STL/blob/b3504262fe51b28ca270aa2e05146984ef758428/stl/CMakeLists.txt#L237-L239 builds those files into the static lib and import lib, but we also want to build this file into the DLL.

Reply:
That's what I tried first; sadly the old msbuild system has a hard restriction that a given `.nativeproj` / `.vcxproj` only compiles a given `.cpp` into one resulting binary. Since one `.nativeproj` produces both "P" and its import lib, it isn't possible to get msbuild to do that (at least with our divisional targets).

I could have created a new .cpp file only included by the old build system which merely included `syserror_import_lib.cpp`, but the `_DLL` test seemed less difficult to follow for future Bill.

Perhaps this could get a transition comment?

Reply:
(See https://github.com/microsoft/STL/pull/457/commits/47b7a92d8296b03fe61ef7761eb9ec17a67102a0 -- at first I tried just including into both binaries but was forced to back that out there because msbuild rejected it)

Reply:
I would accept a `TRANSITION` comment explaining that when the legacy build system is eradicated, we should clean this up.

Reply:
Will do